ibm/resource_ibm_pi_dhcp.go

```diff
@@ -0,0 +1,236 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/errors"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/IBM-Cloud/power-go-client/power/client/p_cloud_service_d_h_c_p"
+)
+
+const (
+	PIDhcpStatusBuilding = "Building"
+	PIDhcpStatusActive   = "ACTIVE"
+	PIDhcpDeleting       = "Deleting"
+	PIDhcpDeleted        = "Deleted"
+	PIDhcpId             = "pi_dhcp_id"
+	PIDhcpStatus         = "pi_dhcp_status"
```

Comment:
All PI resources and datasources we followed some guideline all required and optional arguments starts with pi_XXXX and computed without any pi_ (both for resources and datasources)